File: AirPurity.API.Data/Entities/Notification.cs
namespace AirPurity.API.Data.Entities
{
    public class Notification : BaseModel
    {
        public Notification()
        {
            this.NotificationSubjects = new HashSet<NotificationSubject>();
        }

        public string UserEmail { get; set; }
        public int CityId { get; set; }
        public virtual City City { get; set; }
        public int StationId { get; set; }
        public virtual Station Station { get; set; }
        public virtual ICollection<NotificationSubject> NotificationSubjects { get; set; }
        public int? IndexLevelId { get; set; }
        public int? LastIndexLevelId { get; set; }

    }
}

File: API/ClientApp/src/app/models/notification.ts
import { NotificationSubject } from "./notificationSubject";

export class Notification{
    userEmail: string;
    cityId: number;
    stationId: number;
    indexLevelId: number | null;
    notificationSubjects : NotificationSubject[]

    public constructor(init?: Partial<Notification>){
        if(init){
            init.notificationSubjects = init.notificationSubjects?.filter(x => !isNaN(x.indexLevelId) && x.paramCode?.length > 0) ?? [];
            Object.assign(this, init);
        }
        else{
            this.notificationSubjects = [];
        }
    }
}